File: app/backend/src/config/aws.js
/**
 * Shared AWS SDK client instances.
 * EC2 instances use their IAM instance profile for authentication
 * (aws_iam_instance_profile.app → aws_iam_role.app_instance).
 *
 * AWS services used:
 *   - aws_iam_role.app_instance
 *   - aws_iam_instance_profile.app
 *   - aws_iam_role_policy_attachment.ssm
 *   - aws_iam_role_policy_attachment.ecr_readonly
 */

const { S3Client }              = require('@aws-sdk/client-s3');
const { SQSClient }             = require('@aws-sdk/client-sqs');
const { SNSClient }             = require('@aws-sdk/client-sns');
const { DynamoDBClient }        = require('@aws-sdk/client-dynamodb');
const { DynamoDBDocumentClient } = require('@aws-sdk/lib-dynamodb');
const { SSMClient }             = require('@aws-sdk/client-ssm');
const { SecretsManagerClient }  = require('@aws-sdk/client-secrets-manager');

const REGION = process.env.AWS_REGION || 'us-east-1';

const s3     = new S3Client({ region: REGION });
const sqs    = new SQSClient({ region: REGION });
const sns    = new SNSClient({ region: REGION });
const dynamo = DynamoDBDocumentClient.from(new DynamoDBClient({ region: REGION }));
const ssm    = new SSMClient({ region: REGION });
const secrets = new SecretsManagerClient({ region: REGION });

module.exports = { s3, sqs, sns, dynamo, ssm, secrets };

File: app/backend/src/config/aws.py
import os

import boto3

AWS_REGION = os.getenv("AWS_REGION", "us-east-1")
AWS_ENDPOINT_URL = os.getenv("AWS_ENDPOINT_URL", None)  # For LocalStack / dev overrides

_boto_kwargs = {
    "region_name": AWS_REGION,
}
if AWS_ENDPOINT_URL:
    _boto_kwargs["endpoint_url"] = AWS_ENDPOINT_URL


def get_s3_client():
    return boto3.client("s3", **_boto_kwargs)


def get_sqs_client():
    return boto3.client("sqs", **_boto_kwargs)


def get_sns_client():
    return boto3.client("sns", **_boto_kwargs)


def get_secrets_manager_client():
    return boto3.client("secretsmanager", **_boto_kwargs)


# Singleton clients for reuse within a process
s3_client = get_s3_client()
sqs_client = get_sqs_client()
sns_client = get_sns_client()
secrets_manager_client = get_secrets_manager_client()

S3_BUCKET_NAME = os.getenv("S3_BUCKET_NAME", "insureco-documents")
SQS_QUEUE_URL = os.getenv("SQS_QUEUE_URL", "")
SNS_TOPIC_ARN = os.getenv("SNS_TOPIC_ARN", "")
